workflows/rnavar.nf

```diff
@@ -0,0 +1,431 @@
+            // they can either or both be optional
+            known_sites     = dbsnp.concat(known_indels).collect()
+            known_sites_tbi = dbsnp_tbi.concat(known_indels_tbi).collect()
+
+            ch_interval_list_recalib = ch_interval_list.map{ meta, bed -> [bed] }.flatten()
+            splitncigar_bam_bai.combine(ch_interval_list_recalib)
+                .map{ meta, bam, bai, interval -> [ meta, bam, bai, interval]
+            }.set{splitncigar_bam_bai_interval}
+
+            GATK4_BASERECALIBRATOR(
+                splitncigar_bam_bai_interval,
+                PREPARE_GENOME.out.fasta,
+                PREPARE_GENOME.out.fai,
+                PREPARE_GENOME.out.dict,
+                known_sites,
+                known_sites_tbi
+            )
+            ch_bqsr_table   = GATK4_BASERECALIBRATOR.out.table
+
+            // Gather QC reports
+            ch_reports  = ch_reports.mix(ch_bqsr_table.map{ meta, table -> table})
+            ch_versions     = ch_versions.mix(GATK4_BASERECALIBRATOR.out.versions.first().ifEmpty(null))
+
+            // MODULE: ApplyBaseRecalibrator from GATK4
+            bam_applybqsr       = splitncigar_bam_bai.join(ch_bqsr_table, by: [0])
+            bam_recalibrated_qc = Channel.empty()
+
+            ch_interval_list_applybqsr = ch_interval_list.map{ meta, bed -> [bed] }.flatten()
+            bam_applybqsr.combine(ch_interval_list_applybqsr)
+                .map{ meta, bam, bai, table, interval -> [ meta, bam, bai, table, interval]
+            }.set{applybqsr_bam_bai_interval}
+
+            RECALIBRATE(
+                params.skip_qc,
+                applybqsr_bam_bai_interval,
+                PREPARE_GENOME.out.dict,
+                PREPARE_GENOME.out.fai,
+                PREPARE_GENOME.out.fasta
+            )
+
+            bam_variant_calling = RECALIBRATE.out.bam
+            bam_recalibrated_qc = RECALIBRATE.out.qc
+
+            // Gather QC reports
+            ch_reports          = ch_reports.mix(RECALIBRATE.out.qc.collect{it[1]}.ifEmpty([]))
+            ch_versions         = ch_versions.mix(RECALIBRATE.out.versions.first().ifEmpty(null))
+        } else {
+            bam_variant_calling = splitncigar_bam_bai
+        }
+
+        // MODULE: HaplotypeCaller from GATK4
+        interval_flag = params.no_intervals
+        // Run haplotyper even in the absence of dbSNP files
+        if (!params.dbsnp){
+            dbsnp = []
+            dbsnp_tbi = []
+        }
+
+        haplotypecaller_vcf = Channel.empty()
+
+        haplotypecaller_interval_bam = bam_variant_calling.combine(ch_interval_list_split)
+            .map{ meta, bam, bai, interval_list ->
+                new_meta = meta.clone()
+                new_meta.id = meta.id + "_" + interval_list.baseName
+                new_meta.sample = meta.id
+                [new_meta, bam, bai, interval_list]
+            }
+
+        GATK4_HAPLOTYPECALLER(
```

Comment:
The commenting drops off in this end of the script. Could do with some more description in general as to what is going on in these parts of the pipeline

Reply:
Yes, I overlooked the comments at the pipeline
Now they are added at each process stage.